pkg/client/pods.go

```diff
@@ -73,6 +74,11 @@ func (c *pods) Delete(name string) error {
 	return c.r.Delete().Namespace(c.ns).Resource("pods").Name(name).Do().Error()
 }
 
+// Same as above but with pod termination grace period specified.
+func (c *pods) DeleteGracefully(name string, gracePeriod uint64) error {
```

Comment:
You mean converting DeleteOptions object to appropriate version, encode it and attach as a body to the request, do I understand correctly?